### DISCUSSION TITLE
Please update the Stackblitz template

### DISCUSSION THREAD

**Discussion by mcmxcdev** [melt-ui/melt-ui#814] [Ideas]
Please update https://stackblitz.com/edit/melt-ui-playground to include `lucide-svelte` and `@internationalized/date` libraries by default. 

This will help everyone save time when creating bug reproduction cases.

**Comment by mcmxcdev** [comment]
`@melt-ui/svelte` is also horribly outdated in the Stackblitz template, leading to `TypeError: Cannot read properties of undefined (reading 'root')` when trying to use e.g. "Tags Input "component. I assume that component was created after v0.22.0.

This is what version is used in `pnpm-lock.yaml`:
```
  '@melt-ui/svelte':
    specifier: ^0.22.0
    version: 0.22.0(svelte@4.0.5)
```

**Comment by TGlide** [comment]
Updated the recommendation to SvelteLab's template instead

**Comment by mcmxcdev** [comment]
I find the decision to switch from stackblitz (10k stars) to sveltelab (200 stars) very questionable.

https://www.sveltelab.dev/?t=melt_ui should use pnpm instead of npm to install deps. No lockfile is showing up after installation which makes it hard to know which versions are installed.

`lucide-svelte` and `@internationalized/date` are again not part of the pre-installed deps.

Also, again possibly outdated deps are used, like `"@melt-ui/svelte": "^0.30.0"` which might or might not give you latest.

**Reply by TGlide** [reply]
> I find the decision to switch from stackblitz (10k stars) to sveltelab (200 stars) very questionable.

From my experience SvelteLab is the most used web environment to share SvelteKit examples. It's also much more straightforward, than to create a GitHub action that automatically updates the template.

> https://www.sveltelab.dev/?t=melt_ui should use pnpm instead of npm to install deps.

Why? Don't get me wrong, I love pnpm and use it everywhere, but you can use npm with Melt UI. Or yarn, bun, etc.

> No lockfile is showing up after installation which makes it hard to know which versions are installed.

I think it hides lockfiles by default, but they are there, I think. @paoloricciuti can you confirm?

> lucide-svelte and @internationalized/date are again not part of the pre-installed deps.

You don't *need* either of those packages to run Melt. Lucide is only used by our examples, and the date package is only for the date picker components. But I do agree it'd make it a bit easier to create repros. We can improve it in the future.

> Also, again possibly outdated deps are used, like "@melt-ui/svelte": "^0.30.0" which might or might not give you latest.

On a fresh install (which is the case), it'll install the latest version that is not a major update. This is quite fine until v1

**Reply by paoloricciuti** [reply]
It should not hide the lockfile by default...will look into this is confirmed and eventually fix this bug. Any other specific issue with sveltelab except for "it has less stars than stackblitz"?

**Reply by mcmxcdev** [reply]
> From my experience SvelteLab is the most used web environment to share SvelteKit examples. It's also much more straightforward, than to create a GitHub action that automatically updates the template.

It's much less battle-tested and mature at this point, but let's see how the velocity of it develops.

> Why? Don't get me wrong, I love pnpm and use it everywhere, but you can use npm with Melt UI. Or yarn, bun, etc.

Installing with npm takes unnecessarily longer; multiply that by the time each consumer of melt-ui spends on doing a repro case and waiting for the environment to fully load. It's also a lot less strict than pnpm, but it's personal choice after all.

> You don't need either of those packages to run Melt. Lucide is only used by our examples, and the date package is only for the date picker components. But I do agree it'd make it a bit easier to create repros. We can improve it in the future.

Yes, neither of them are necessary, but most consumers will just copy and paste the examples, which then end up breaking in the repro templates. But we do agree on this one it seems!

> On a fresh install (which is the case), it'll install the latest version that is not a major update. This is quite fine until v1

Like mentioned in [a comment above](https://github.com/melt-ui/melt-ui/discussions/814#discussioncomment-7888692), I ended up with an old melt-ui version like 0.22.0, even though it was supposed to pull latest in stackblitz as well. Either locking the versions to fixed or using latest like ^0.66.3 would already help there.

***

This discussion ended up sounding much more hostile than I wanted it to, sorry! Both of you, thanks for your great work :rocket:

**Reply by AdrianGonz97** [reply]
It looks like the lock-file is included in the template: https://github.com/SvelteLab/SvelteLab/tree/main/src/lib/default_project_files/melt_ui

Deleting it from the template here should install the latest versions on initialization.

edit: Opened a PR for this as it's a quick change https://github.com/SvelteLab/SvelteLab/pull/295

**Reply by TGlide** [reply]
> Installing with npm takes unnecessarily longer;

Is this true with webcontainers as well? I would assume they wouldn't have anything cached to make install times faster. Otherwise we could look into it.

> even though it was supposed to pull latest in stackblitz as well.

The stackblitz template had a lockfile, IIRC

> This discussion ended up sounding much more hostile than I wanted it to, sorry! 

No worries, happens all the time. I also apologize if I sounded hostile, and thank you for voicing all your concerns to help make Melt better 🙂

**Reply by mcmxcdev** [reply]
> Deleting it from the template here should install the latest versions on initialization.

Installing without lockfile is risky too, since then the template could end up being in a broken state whenever deps don't work with each other (especially with potential breaking changes for something like v0 that melt-ui is).

I understand the maintenance burden, but updating the lockfile here and there manually to a state that is confirmed to be working would probably be ideal.

> Is this true with webcontainers as well? I would assume they wouldn't have anything cached to make install times faster. Otherwise we could look into it.

Might be biased, but here are benchmarks: https://github.com/pnpm/pnpm#benchmark

**Reply by TGlide** [reply]
> I understand the maintenance burden, but updating the lockfile here and there manually to a state that is confirmed to be working would probably be ideal.

I'll have to look into it. Automation is the best way forward, if possible. Need to look at how other repos do it

> Might be biased, but here are benchmarks

lol I love pnpm. @paoloricciuti any chance we can update pnpm to be the default in the melt template?

**Reply by paoloricciuti** [reply]
We default to npm for sveltelab because it's simpler for the users...but I might look into way to customize this, I'm a `pnpm` lover too and would love to use it

**Comment by TGlide** [comment]
Solved, we updated the Stackblitz template, added a link to it ([melt-ui.com/repro](https://melt-ui.com/repro)), and updated the bug report template. All thanks to @huntabyte!
